File: JAVA/Assignment 15/Program3.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write a java program whihc Fibonacci series.                    //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 8 Aug 2026                                                                   //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class FibonacciSeries
{
    public int FindFibonacci(int iNo)
    {
        if(iNo < 0) return -1;
        if(iNo == 0) return 0;
        if(iNo == 1 || iNo == 2) return 1;

        int first = 1, second = 1, current = 2; 

        for(int i = 2 ; i < iNo ; i++)
        {
            current = first + second;
            first = second;
            second = current;   
        }
        return current;
    }
}

class Program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter Number for Fibonacci Series : ");
        int no = sobj.nextInt();

        FibonacciSeries fobj = new FibonacciSeries();

        int iRet = fobj.FindFibonacci(no);
        if(iRet == -1)
        {
            System.out.println("Enter Position Integer!");
            return;
        }
        System.out.println(iRet);
    }
}

File: C/Assignment32/Program3.c
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// File name :   Program3.c
// Description : Write a program which accept one number & position from user & on that bit. Return modified number.
// Author :      Pallavi Omprakash Malewar
// Date :        25/07/2026
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

/*

    Input :  10
             3

    Output : 14

*/

#include<stdio.h>

typedef unsigned int UINT;
typedef int BOOL;

#define TRUE 1
#define FALSE 0

UINT ONBit(UINT iNo, int iPos)
{
    if(iPos < 1 || iPos > 32) return FALSE;
    
    int iMask = 1 << (iPos - 1);
    return iNo | iMask;
}

int main()
{
    int iNo = 0, iPos = 0;
    int iRet = 0;

    printf("Enter the number : ");
    scanf("%d", &iNo);

    printf("Enter Position : ");
    scanf("%d", &iPos);

    iRet = ONBit(iNo, iPos);

    printf("%d\n", iRet);

    return 0;
}